src/App/Fossa/Analyze/ScanSummary.hs

```diff
@@ -0,0 +1,244 @@
+
+getBinaryIdentifier :: SourceUnit -> [Text]
+getBinaryIdentifier srcUnit = maybe [] (srcUserDepName <$>) (userDefinedDeps =<< additionalData srcUnit)
+
+srcUnitToScanCount :: Result (Maybe SourceUnit) -> ScanCount
+srcUnitToScanCount (Failure _ _) = ScanCount 1 0 0 1 0
+srcUnitToScanCount (Success _ Nothing) = ScanCount 0 0 0 0 0
+srcUnitToScanCount (Success wg (Just _)) = ScanCount 1 0 1 0 (countWarnings wg)
+
+summarizeSrcUnit ::
+  (Has Diag.Diagnostics sig m, Has Logger sig m) =>
+  Text ->
+  (Maybe (SourceUnit -> [Text])) ->
+  Result (Maybe SourceUnit) ->
+  m ()
+summarizeSrcUnit analysisHeader maybeGetter (Success wg (Just unit)) = do
+  logInfo $ successColorCoded wg $ pretty (listSymbol <> analysisHeader) <> renderSucceeded wg
+  case maybeGetter <*> (Just unit) of
+    Just txts -> logInfoVsep $ itemize ("  *" <> listSymbol) pretty txts
+    Nothing -> pure ()
+summarizeSrcUnit analysisHeader _ (Failure _ _) =
+  logInfo . failColorCoded $
+    (annotate bold . pretty $ analysisHeader)
+      <> renderFailed
+summarizeSrcUnit _ _ _ = pure ()
+
+summarizeProjectScan :: DiscoveredProjectScan -> Doc AnsiStyle
+summarizeProjectScan (Scanned dpi (Failure _ _)) = failColorCoded $ renderDiscoveredProjectIdentifier dpi <> renderFailed
+summarizeProjectScan (Scanned _ (Success wg pr)) = successColorCoded wg $ renderProjectResult pr <> renderSucceeded wg
+summarizeProjectScan (SkippedDueToProvidedFilter dpi) = renderDiscoveredProjectIdentifier dpi <> renderSkipped
+summarizeProjectScan (SkippedDueToDefaultProductionFilter dpi) = renderDiscoveredProjectIdentifier dpi <> renderSkipped
+
+---------- Rendering Helpers
+
+logInfoVsep :: (Has Logger sig m) => [Doc AnsiStyle] -> m ()
+logInfoVsep = traverse_ logInfo
+
+renderDiscoveredProjectIdentifier :: DiscoveredProjectIdentifier -> Doc AnsiStyle
+renderDiscoveredProjectIdentifier dpi = renderProjectPathAndType (dpiProjectType dpi) (dpiProjectPath dpi)
+
+renderProjectResult :: ProjectResult -> Doc AnsiStyle
+renderProjectResult pr = renderProjectPathAndType (projectResultType pr) (projectResultPath pr)
+
+renderProjectPathAndType :: DiscoveredProjectType -> Path Abs Dir -> Doc AnsiStyle
+renderProjectPathAndType pt path =
+  (annotate bold . pretty $ projectTypeToText pt)
+    <> pretty (" project in " :: Text)
+    <> pretty (toText path)
+
+successColorCoded :: [EmittedWarn] -> Doc AnsiStyle -> Doc AnsiStyle
+successColorCoded ew = if countWarnings ew == 0 then annotate (color Green) else annotate (color Yellow)
+
+failColorCoded :: Doc AnsiStyle -> Doc AnsiStyle
+failColorCoded = annotate (color Red)
+
+renderSkipped :: Doc ann
+renderSkipped = pretty (": skipped" :: Text)
+
+renderSucceeded :: [EmittedWarn] -> Doc ann
+renderSucceeded ew =
+  if countWarnings ew == 0
+    then pretty (": succeeded" :: Text)
+    else pretty (": succeeded with " :: Text) <> pretty (show $ countWarnings ew) <> pretty (": warning" :: Text)
+
+renderFailed :: Doc ann
+renderFailed = pretty (": failed" :: Text)
```

Comment:
`Doc ann` has an instance for `IsString`, there's no need to use `Text` here:
```suggestion
renderFailed = ": failed"
```

You should replace all instances of `pretty ("foo" :: Text)` with `"foo"`

Reply:
Addressed in https://github.com/fossas/fossa-cli/pull/790/commits/b558311435073a8500f63e5365ea3304ffc90d31